### ISSUE TITLE
how to use OptimizeImageLayers

### ISSUE THREAD

**Issue by 2016xjtuzyt** [ImageMagick/ImageMagick#896]
hi，
my ImageMagic version is ImageMagick 6.7.8-9 in Centos 6.5,
I use this command "convert test.gif -fuzz 15% -layers Optimize result.gif" and get the small animation gif.
how to convert this command to C++ code?
when I write the this code:
    Image img;
    img.read("animation.gif");
    Image OptimizeImageLayers(img);
    img.write("out.gif");
the out.gif is one frame, not animation.
Thanks!

**Comment by khanhas** [comment]
`Magick::read` method is only able to read one frame. If you want to read multiple frames, use `readImages` to put them in a vector. So later, you can use STL interface to edit them.
```cpp
std::vector<Magick::Image> frameList;
Magick::readImages(&frameList, "animation.gif");
```

**Comment by 2016xjtuzyt** [comment]
@khanhas thanks for your reply. i can't find optimizeImagelayers in STL.h. this function is define in imageMagick7.xx. when i install imageMagick7.0.7,there is something wrong as follows:
MagickCore/.libs/libMagickCore-7.Q16HDRI.so: undefined reference to `WebPMemoryWriterClear'
collect2: error: ld returned 1 exit status
my system is Centos6.5, can you help me?

**Comment by khanhas** [comment]
Sorry, I do not use Linux right now so I have no idea how to fix that.
Seems like there's something wrong with WebP library.

**Comment by 2016xjtuzyt** [comment]
@khanhas thank you very much.

**Comment by magiclen** [comment]
I got the same message `undefined reference to `WebPMemoryWriterClear'` when I compiled ImageMagick7.0.7-18 on CentOS. But ImageMagick7.0.7-11 is fine. It can be compiled and used on CentOS successfully.

**Comment by urban-warrior** [comment]
We added a patch to the ImageMagick master branch to check the version of the libwebpmux library. Let us know if the patch fixes the linking issue under CentOS 6.5.

**Comment by magiclen** [comment]
@urban-warrior I've tried ImageMagick from the master branch, and it can be successfully compiled now!
